awswrangler/athena/_spark.py

```diff
@@ -0,0 +1,210 @@
+
+    _wait_calculation_execution(
+        calculation_execution_id=calculation_execution_id,
+        boto3_session=boto3_session,
+    )
+
+    response = client_athena.get_calculation_execution(
+        CalculationExecutionId=calculation_execution_id,
+    )
+    # TODO: process results by content-types
+    # result_type = response["Result"]["ResultType"]
+    # result_s3_uri = response["Result"]["ResultS3Uri"]
+    # if result_type == "application/vnd.aws.athena.v1+json":
+    #    return s3.read_json(path=result_s3_uri)
+    return response
+
+
+def create_spark_session(
+    workgroup: str,
+    notebook_id: Optional[str] = None,
+    notebook_version: str = "Athena notebook version 1",
+    coordinator_dpu_size: int = 1,
+    max_concurrent_dpus: int = 5,
+    default_executor_dpu_size: int = 1,
+    additional_configs: Optional[Dict[str, Any]] = None,
+    idle_timeout: int = 15,
+    boto3_session: Optional[boto3.Session] = None,
+) -> str:
+    """
+    Create session and wait until ready to accept calculations.
+
+    Parameters
+    ----------
+    workgroup : str
+    notebook_id : str, optional
+    notebook_version : str, optional
+    coordinator_dpu_size : int, optional
+    max_concurrent_dpus : int, optional
+    default_executor_dpu_size: int, optional
+    additional_configs : Dict[str, Any], optional
+    idle_timeout : int, optional
+    boto3_session : boto3.Session(), optional
+
+    Returns
+    -------
+        Session id
+    """
+    client_athena = _utils.client(service_name="athena", session=boto3_session)
+    engine_configuration: "EngineConfigurationTypeDef" = {
+        "CoordinatorDpuSize": coordinator_dpu_size,
+        "MaxConcurrentDpus": max_concurrent_dpus,
+        "DefaultExecutorDpuSize": default_executor_dpu_size,
+    }
+    if additional_configs:
+        engine_configuration["AdditionalConfigs"] = additional_configs
+    response = client_athena.start_session(
+        WorkGroup=workgroup,
+        EngineConfiguration=engine_configuration,
+        # NotebookId=notebook_id,
+        # NotebookVersion=notebook_version if notebook_id else None,
+        SessionIdleTimeoutInMinutes=idle_timeout,
+    )
+    _logger.info("Session info:\n%s", response)
+    session_id: str = response["SessionId"]
+    # Wait for the session to reach IDLE state to be able to accept calculations
+    _wait_session(
+        session_id=session_id,
+        boto3_session=boto3_session,
+    )
+    return session_id
+
+
+def run_spark_calculation(
+    code: str,
+    workgroup: str,
+    session_id: Optional[str] = None,
+    notebook_id: Optional[str] = None,
+    notebook_version: str = "Athena notebook version 1",
+    coordinator_dpu_size: int = 1,
+    max_concurrent_dpus: int = 5,
+    default_executor_dpu_size: int = 1,
+    additional_configs: Optional[Dict[str, Any]] = None,
+    idle_timeout: int = 15,
+    boto3_session: Optional[boto3.Session] = None,
+) -> "GetCalculationExecutionResponseTypeDef":
```

Comment:
So far, we haven't been returning any of the boto3 stub types from our public facing functions. Our reasoning was that for customers who don't use the boto3 stubs, it'll be more helpful for the return type to be returned as something like `Dict[str, Any]` than just `Any`.

Reply:
Yeah good point, will change back to dicts